src/main/java/com/gmail/inverseconduit/bot/Program.java

```diff
@@ -51,66 +43,28 @@
     public Program(ChatInterface chatInterface) throws IOException {
         LOGGER.finest("Instantiating Program");
         this.chatInterface = chatInterface;
-        this.bot = new DefaultBot(chatInterface);
-        this.interactionBot = new InteractionBot(chatInterface);
-
-        JavaDocAccessor tmp;
-        //better not get ExceptionInInitializerError
-        try {
-            tmp = new JavaDocAccessor(config.getJavadocsDir());
-        } catch(IOException ex) {
-            LOGGER.log(Level.WARNING, "Couldn't initialize Javadoc accessor.", ex);
-            tmp = null;
-        }
-        this.javaDocAccessor = tmp;
-
-        chatInterface.subscribe(bot);
-        chatInterface.subscribe(interactionBot);
+        bots.add(new DefaultBot(chatInterface));
+        bots.add(new InteractionBot(chatInterface));
+        bots.forEach(chatInterface::subscribe);
         LOGGER.info("Basic component setup complete");
     }
 
     /**
-     * This is where the beef happens. Glue all the stuff together here
+     * Here the injected chatInterface is used to join the rooms specified in
+     * bot.properties.
+     * Additionally all bots that were created on startup are started.
      */
     public void startup() {
         LOGGER.info("Beginning startup process");
-        bindDefaultCommands();
         for (Integer room : config.getRooms()) {
             // FIXME: isn't always Stackoverflow
             chatInterface.joinChat(new SeChatDescriptor.DescriptorBuilder(SESite.STACK_OVERFLOW).setRoom(() -> room).build());
         }
-        bot.start();
-        interactionBot.start();
+        bots.forEach(ChatWorker::start);
         LOGGER.info("Startup completed.");
     }
 
-    private void bindDefaultCommands() {
-        bindNumberCommand();
-        bindJavaDocCommand();
-    }
-
-    private void bindNumberCommand() {
-        final Pattern p = Pattern.compile("^\\d+$");
```

Comment:
Considering how we're using this pattern for anchored matching, we could make it `"\\d+"` and replace the line down there with `matcher.matches()`.